content/blog/how-to-locate-elements-in-playwright.md

```diff
@@ -0,0 +1,183 @@
+
+You can also pick a locator when using the standalone test generator by clicking the explore button.
+
+```bash
+npx playwright open
+```
+
+![test generator generating a getByRole locator for a link](https://res.cloudinary.com/debsobrien/image/upload/v1669795104/debbie.codes/blog/2022/locator-link_hz7icx.png)
+
+### Chaining locators
+
+If your site contains duplicate links, for example two about links, one in the header and one in the footer, then the test generator will create a unique locator for each one. It does this by looking at the next accessible ancestor to the link and chains the locators.
+
+```js
+getByRole('navigation').getByRole('link', { name: 'About' })
+```
+
+![test generator highlighting the about link in the header](https://res.cloudinary.com/debsobrien/image/upload/v1669795104/debbie.codes/blog/2022/locator-chaining-header_hacas1.png)
+
+
+For the about link in the footer the test generator again looks at the next accessible ancestor to the link and chains the two [`getByRole()`](https://playwright.dev/docs/locators#locate-by-role) locators ensuring that there is only one instance for this locator.
+
+```js
+getByRole('contentinfo').getByRole('link', { name: 'About' })
+```
+
+![test generator highlighting the footer link](https://res.cloudinary.com/debsobrien/image/upload/v1669795104/debbie.codes/blog/2022/locator-chaining-footer_ogczv7.png)
+
+### Filtering locators
+
+If the test generator can't give you a unique locator when chaining it will use the [`filter()`](https://playwright.dev/docs/locators#filtering-locators) method to ensure the locator is unique. For example we might have two lists that contain topics for blog posts. The first list is used to filter the blog posts and the second list is used to display the topics for each post. The test generator will use the [`filter()`](https://playwright.dev/docs/locators#filtering-locators) method to ensure the locator is unique.
+
+```js
+getByRole('list')
+  .filter({ hasText: 'architecturedev reljamstackjavascriptlifestylementoringmotivationnuxtperformance' })
+  .getByRole('link', { name: 'architecture' })
+```
+
+![test generator generating a list locator with filter method](https://res.cloudinary.com/debsobrien/image/upload/v1669795104/debbie.codes/blog/2022/locator-filtering_hvlbco.png)
+
+### Improving your locators
+
+You may notice that filtering by text can sometimes give you a really ugly locator, like in the example above, and may cause you issues later, especially if you add another topic to the filters. If you want to improve this locator you first need to improve the HTML. For example you can add an aria-label to the first `<ul>` element with the value of topics. This not only helps improve the accessability of the page but also allows Playwright to locate by a role selector.
```

Comment:
I think we should first explain that you can tweak locator to make it better in this case. For example, replace long `hasText` with shorter one that is resilient. And only as a last resort, you would go and change the page itself.